sound/soc/codecs/rt5682-sdw.c

```diff
@@ -400,6 +400,11 @@ static int rt5682_io_init(struct device *dev, struct sdw_slave *slave)
 
 	pm_runtime_get_noresume(&slave->dev);
 
+	if (rt5682->first_hw_init) {
+		regcache_cache_only(rt5682->regmap, false);
+		regcache_cache_bypass(rt5682->regmap, true);
+	}
+
 	while (loop > 0) {
 		regmap_read(rt5682->regmap, RT5682_DEVICE_ID, &val);
```

Comment:
@bardliao Possibly dumb question, but wouldn't it be better to declare DEVICE_ID as volatile? Or is there some reason this can't be done at this point.

Reply:
> @bardliao Possibly dumb question, but wouldn't it be better to declare DEVICE_ID as volatile? Or is there some reason this can't be done at this point.

Yes, RT5682_DEVICE_ID is volatile and it is the problem. We will never read the volatile register from cache, and regmap_read will return `-EBUSY` when regcache_cache_only is true.